src/BSON/Decimal128.c

```diff
@@ -0,0 +1,205 @@
+
+
+	zend_replace_error_handling(EH_THROW, phongo_exception_from_phongo_domain(PHONGO_ERROR_INVALID_ARGUMENT), &error_handling TSRMLS_CC);
+	intern = Z_DECIMAL128_OBJ_P(getThis());
+
+	if (zend_parse_parameters(ZEND_NUM_ARGS() TSRMLS_CC, "s", &value, &value_len) == FAILURE) {
+		zend_restore_error_handling(&error_handling TSRMLS_CC);
+		return;
+	}
+	zend_restore_error_handling(&error_handling TSRMLS_CC);
+
+	if ( ! bson_decimal128_from_string(value, &intern->decimal)) {
+		phongo_throw_exception(PHONGO_ERROR_INVALID_ARGUMENT TSRMLS_CC, "Error parsing decimal string: %s", value);
+		return;
+	}
+}
+/* }}} */
+
+/* {{{ proto void Decimal128::__toString()
+    */
+PHP_METHOD(Decimal128, __toString)
+{
+	php_phongo_decimal128_t *intern;
+	char                     outbuf[BSON_DECIMAL128_STRING] = "";
+
+	intern = Z_DECIMAL128_OBJ_P(getThis());
+
+	if (zend_parse_parameters_none() == FAILURE) {
+		return;
+	}
+
+	bson_decimal128_to_string(&intern->decimal, outbuf);
+
+	PHONGO_RETURN_STRING(outbuf);
+}
+/* }}} */
+
+
+/* {{{ BSON\Decimal128 */
+
+ZEND_BEGIN_ARG_INFO_EX(ai_Decimal128___construct, 0, 0, 2)
+	ZEND_ARG_INFO(0, value)
+ZEND_END_ARG_INFO();
+
+ZEND_BEGIN_ARG_INFO_EX(ai_Decimal128___toString, 0, 0, 0)
+ZEND_END_ARG_INFO();
+
+
+static zend_function_entry php_phongo_decimal128_me[] = {
+	PHP_ME(Decimal128, __construct, ai_Decimal128___construct, ZEND_ACC_PUBLIC|ZEND_ACC_FINAL)
+	PHP_ME(Decimal128, __toString, ai_Decimal128___toString, ZEND_ACC_PUBLIC|ZEND_ACC_FINAL)
+	PHP_ME(Manager, __wakeUp, NULL, ZEND_ACC_PUBLIC)
+	PHP_FE_END
+};
+
+/* }}} */
+
+
+/* {{{ php_phongo_decimal128_t object handlers */
+static void php_phongo_decimal128_free_object(phongo_free_object_arg *object TSRMLS_DC) /* {{{ */
+{
+	php_phongo_decimal128_t *intern = Z_OBJ_DECIMAL128(object);
+
+	zend_object_std_dtor(&intern->std TSRMLS_CC);
+
+#if PHP_VERSION_ID < 70000
+	efree(intern);
+#endif
+} /* }}} */
+
+phongo_create_object_retval php_phongo_decimal128_create_object(zend_class_entry *class_type TSRMLS_DC) /* {{{ */
+{
+	php_phongo_decimal128_t *intern = NULL;
+
+	intern = PHONGO_ALLOC_OBJECT_T(php_phongo_decimal128_t, class_type);
+
+	zend_object_std_init(&intern->std, class_type TSRMLS_CC);
+	object_properties_init(&intern->std, class_type);
+
+#if PHP_VERSION_ID >= 70000
+	intern->std.handlers = &php_phongo_handler_decimal128;
+
+	return &intern->std;
+#else
+	{
+		zend_object_value retval;
+		retval.handle = zend_objects_store_put(intern, (zend_objects_store_dtor_t) zend_objects_destroy_object, php_phongo_decimal128_free_object, NULL TSRMLS_CC);
+		retval.handlers = &php_phongo_handler_decimal128;
+
+		return retval;
+	}
+#endif
+} /* }}} */
+
+HashTable *php_phongo_decimal128_get_debug_info(zval *object, int *is_temp TSRMLS_DC) /* {{{ */
+{
+	php_phongo_decimal128_t *intern;
+	char                  outbuf[BSON_DECIMAL128_STRING] = "";
+#if PHP_VERSION_ID >= 70000
+	zval                  retval;
+#else
+	zval                  retval = zval_used_for_init;
```

Comment:
Is this not the wrong way around? It looks odd to set a zval to a static value, but not do that for PHP 7+